File: 64Bit/extensions/01 charts/ws_tbar.lua
local pageCharts = std.ToolBarPage.new("Charts")

local btnScatter = std.Button.new{
	title = "scatter", 
	py = ACTIVEDIR.."/scatter.py", 
	img = ACTIVEDIR.."/icons/scatter.png"
}

local btnBubble = std.Button.new{
	title = "bubble", 
	py = ACTIVEDIR.."/bubble.py", 
	img = ACTIVEDIR.."/icons/bubble.png"
}

local btnHistogram = std.Button.new{
	title = "histogram", 
	py = ACTIVEDIR.."/histogram.py", 
	img = ACTIVEDIR.."/icons/histogram.png"
}

local btnBoxWhisker = std.Button.new{
	title = "box-whisker", 
	py = ACTIVEDIR.."/boxwhisker.py", 
	img = ACTIVEDIR.."/icons/boxwhisker.png"
}

local btnColClusterChrt = std.Button.new{
	title = "column", 
	py = ACTIVEDIR.."/col_cluster.py", 
	img = ACTIVEDIR.."/icons/col_cluster.png"
}

local btnColStackChrt = std.Button.new{
	title = "column-stacked", 
	py = ACTIVEDIR.."/col_stk.py", 
	img = ACTIVEDIR.."/icons/col_stk.png"
}

local btnQQ = std.Button.new{
	title = "Q-Q", 
	py = ACTIVEDIR.."/qqchart.py", 
	img = ACTIVEDIR.."/icons/qqchart.png"
}

local btnPsyChart = std.Button.new{
	title = "psychrometry", 
	py = ACTIVEDIR.."/psychart.py", 
	img = ACTIVEDIR.."/icons/psychart.png"
}

local btnMoody = std.Button.new{
	title = "Moody", 
	py = ACTIVEDIR.."/moody.py", 
	img = ACTIVEDIR.."/icons/moody.png"
}


local btnColCharts = std.HybridButton.new(btnColClusterChrt)
btnColCharts:add(btnColStackChrt)


pageCharts:add(btnScatter)
pageCharts:add(btnBubble)
pageCharts:add(btnColCharts)
pageCharts:add(btnHistogram)
pageCharts:add(btnBoxWhisker)
pageCharts:add(btnQQ)
pageCharts:add(btnPsyChart)
pageCharts:add(btnMoody)

pageCharts:start()

File: 64Bit/extensions/01 charts/ws_tbar.py
import pathlib

import _sci.extension as ext
from _sci.framework import Framework


def run(x):
	pass


if __name__ == "__main__":
	CurFile = str(__file__)
	CurFolder = pathlib.Path(__file__).parent

	i = -1

	btnScatter = ext.Button("scatter", str(CurFolder/"icons/scatter.png"), CurFile, run, i:=i+1)
	btnBubble = ext.Button("bubble", str(CurFolder / "icons/bubble.png"), CurFile, run, i:=i+1)
	btnHistogram = ext.Button("histogram", str(CurFolder/"icons/histogram.png"), CurFile, run, i:=i+1)
	btnBoxWhisker = ext.Button("box-whisker", str(CurFolder / "icons/boxwhisker.png"), CurFile, run, i:=i+1)
	btnColClusterChrt = ext.Button("column", str(CurFolder/"icons/col_cluster.png"), CurFile, run, i:=i+1)
	btnColStackChrt = ext.Button("column-stacked", str(CurFolder/"icons/col_stk.png"), CurFile, run, i:=i+1)

	btnQQ = ext.Button("Q-Q", str(CurFolder/"icons/qqchart.png"), CurFile, run, i:=i+1)
	
	btnPsyChart = ext.Button("psychrometry", str(CurFolder/"icons/psychart.png"), CurFile, run, i:=i+1)
	btnMoody = ext.Button("Moody", str(CurFolder/"icons/moody.png"), CurFile, run, i:=i+1)

	btnColCharts = ext.HybridButton(btnColClusterChrt)
	btnColCharts.add(btnColStackChrt)

	page = ext.Page("Charts")
	page.add(btnScatter)
	page.add(btnBubble)
	page.add(btnColCharts)
	page.add(btnHistogram)
	page.add(btnBoxWhisker)
	page.add(btnQQ)
	page.add(btnPsyChart)
	page.add(btnMoody)

	Framework().ToolBar_AddPage(page)